var/spack/repos/builtin/packages/astyle/package.py

```diff
@@ -11,8 +11,13 @@ class Astyle(Package):
     def install(self, spec, prefix):
 
         with working_dir('src'):
+            # unfortunately we need to edit the makefile in place to set compiler:
+            make_file = join_path(self.stage.source_path,'build','gcc','Makefile')
+            filter_file(r'^CXX\s*=.*', 'CXX=%s' % spack_cxx, make_file)
```

Comment:
If I had to guess why Travis is crashing, it's because of this line. String formatting is a little broken in Python 2. The following two statements are equivalent:
```python
print "%s %s" % "foo", "bar"
print "%s %s" % ("foo", "bar")
```
The problem here is that Python can't distinguish whether `make_file` is an argument to use for string formatting, or a separate argument to `filter_file()`. Here are two solutions:
```python
filter_file(r'^CXX\s*=.*', 'CXX=%s' % (spack_cxx), make_file)
filter_file(r'^CXX\s*=.*', 'CXX=%s'.format(spack_cxx), make_file)
```
The latter is Python 3's syntax, which requires parentheses and prevents this from ever happening. It works in Python 2.6 and 2.7, so it should be fine to integrate into Spack. This is what I prefer to use.